tests/unit/test_service.py

```diff
@@ -381,98 +383,185 @@ def test_process_testplan_request_with_all_attributes(mocker):
-def test_main_with_celery_enabled(mocker):
-    """Test main when Celery is enabled (coverage for line 226)."""
-    import os
+    # Create a mock BackgroundTasks object
+    mock_bg_tasks = mocker.Mock()
 
-    # Ensure environment variable is set to enable Celery (or not set at all)
-    if "USE_CELERY" in os.environ:
-        old_value = os.environ["USE_CELERY"]
-        os.environ.pop("USE_CELERY")
-    else:
-        old_value = None
+    # Call main with test parameters
+    result = process_request(
+        mock_bg_tasks, "url", "test", None, None, None, None, None, None, "json"
+    )
 
-    # Create a mock test
+    # Verify task_manager.execute_task was called with the right parameters
+    mock_task_execute.assert_called_once()
+    assert result == "task-123"
+
+
+def test_process_request_worker_no_output_format(mocker):
+    """Test _process_request_worker when no output format is specified."""
+    # Create mock data
     test_data = TestData(name="test", summary="Test summary")
 
-    # Mock process_test_request to return our test data
+    # Mock process_test_request
     mocker.patch("tmt_web.service.process_test_request", return_value=test_data)
 
-    # Mock format_data to verify it's NOT called directly
-    mock_format = mocker.patch("tmt_web.service.format_data")
-
-    # Mock serialize_data to ensure it IS called and returns serialized data
+    # Mock serialize_data
     mock_serialize = mocker.patch("tmt_web.service.serialize_data", return_value="serialized_data")
 
-    # Call main with test parameters
-    result = main("url", "test", None, None, None, None, None, None, "json")
+    # Call worker with None as format
+    from tmt_web.service import _process_request_worker
 
-    # Verify format_data was NOT called
-    mock_format.assert_not_called()
+    result = _process_request_worker("url", "test", None, None, None, None, None, None, None)
 
-    # Verify serialize_data WAS called with the test data
+    # Verify serialize_data was called instead of format_data
     mock_serialize.assert_called_once_with(test_data)
-
-    # Verify result is from serialize_data
     assert result == "serialized_data"
 
-    # Restore environment if needed
-    if old_value is not None:
-        os.environ["USE_CELERY"] = old_value
+
+def test_task_manager_update_nonexistent_task(mocker):
+    """Test updating a non-existent task in the task manager."""
+    # Mock Valkey client get method to return None (task doesn't exist)
+    mocker.patch.object(task_manager.client, "get", return_value=None)
+
+    # Mock the set method to check it's not called
+    mock_set = mocker.patch.object(task_manager.client, "set")
+
+    # Mock logger to verify warning is called
+    mock_logger = mocker.patch.object(task_manager, "logger")
+
+    # Try to update a non-existent task
+    task_manager.update_task("nonexistent-task-id", "SUCCESS", result="test result")
+
+    # Verify warning was logged
+    mock_logger.warning.assert_called_once_with(
```

Comment:
After it was changed to `fail`, the test should be updated as well.
```suggestion
    mock_logger.fail.assert_called_once_with(
```